### DISCUSSION TITLE
LookupError: 'time-geospatial' is not among the defined enum values. Enum name: featuretype. Possible values: GEOSPATIAL, TIME_GEOSPA..

### DISCUSSION THREAD

**Discussion by rojo1997** [fastapi/sqlmodel#680] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import datetime as dt
from enum import Enum
from sqlmodel import Field, SQLModel, Session, select

class FeatureType(str, Enum):
    GEOSPATIAL = "geospatial"
    TIME_GEOSPATIAL = "time-geospatial"

class Feature(SQLModel, table=True):
    feature_id: str = Field(primary_key=True)
    feature_name: str = Field(index=True, unique=True)
    feature_type: FeatureType

with Session(engine) as session:
    statement = select(Feature).where(Feature.feature_name == feature_name)
    feature: Feature | None = session.exec(statement).first()
```

### Description

After upgrading SQLModel from version 0.0.8 to 0.0.10 I get the following error:
`LookupError: 'time-geospatial' is not among the defined enum values. Enum name:
featuretype. Possible values: GEOSPATIAL, TIME_GEOSPA..`

`CREATE TABLE feature (
	feature_id varchar(255) NOT NULL,
	feature_name varchar(255) NOT NULL,
	feature_type varchar(32) NOT NULL
)
WITH (
	OIDS=TRUE
);`

I am using an AWS Redshift database. The data (Feature object) I am trying to read was inserted and read correctly in version 0.0.8. It seems that when upgrading SQLAlchemy with SQLModel it is requiring the Enum type to exist in the DB instead of just in the Pydantic schema. If this were the case, this upgrade would require a mandatory update in the DDLs of each database table.

It is possible that this issue is only related to SQLAlchemy.

### Operating System

Linux

### Operating System Details

Running in a container Docker image python:3.11-slim-buster

### SQLModel Version

0.0.10

### Python Version

3.11

**Comment by hhartzer** [comment]
I'm seeing this as well. The enum format seems to have changed, which is a pretty breaking change. Not too hard to update, but probably worth looking into some more.

I'm seeing this when going from 0.0.8 to 0.0.9, 0.0.10, or 0.0.11.

**Comment by hhartzer** [comment]
I think this may be hidden now as the question is marked as answered. It seems like something that would be nice for someone else like @tiangolo to have a look at.

**Comment by hohobilly** [comment, marked as answer]
For those who are running into the same issue because your database column has `varchar` type instead of enum:

The root cause is probably from `0.0.9`, a change in enum type checks ordering was introduced. It used to be checking String first and determine the type as `sqlalchemy.types.String`. After `0.0.9` it returns `sa_Enum(enum.Enum)` instead.

Overriding the `sa_type` or `sa_column`  manually worked for me:

```python
from sqlmodel import Column, Field, SQLModel
from sqlmodel.sql.sqltypes import AutoString

class Feature(SQLModel, table=True):
    ...
    # If define sa_type
    feature_type: FeatureType = Field(sa_type=AutoString) # or sqlalchemy.types.String
    # If define sa_column
    feature_type: FeatureType = Field(sa_column=Column(AutoString, ...other kwargs) 
```

Note: `Field` has to be imported from `sqlmodel`, not `pydantic`

**Comment by rojo1997** [comment]
Thanks @hohobilly. It works!!!
